Assets/HoloToolkit/Input/Scripts/InputManager.cs

```diff
@@ -228,54 +235,62 @@ public void HandleEvent<T>(BaseEventData eventData, ExecuteEvents.EventFunction<
             }
 
             // Use focused object when OverrideFocusedObject is null.
-            GameObject focusedObject = (OverrideFocusedObject == null) ? GazeManager.Instance.HitObject : OverrideFocusedObject;
+            GameObject focusedObject = OverrideFocusedObject ?? GazeManager.Instance.HitObject;
 
-            // Send the event to global listeners
+            // Send the event to global listeners.
             for (int i = 0; i < globalListeners.Count; i++)
             {
-                // Global listeners should only get events on themselves, as opposed to their hierarchy
+                // Global listeners should only get events on themselves, as opposed to their hierarchy.
                 ExecuteEvents.Execute(globalListeners[i], eventData, eventHandler);
             }
 
-            // Handle modal input if one exists
+            // Check if the event has already been used by the global listeners.
+            if (eventData.used)
+            {
+                return;
+            }
+
+            // Handle modal input if one exists.
             if (modalInputStack.Count > 0)
             {
                 GameObject modalInput = modalInputStack.Peek();
 
-                // If there is a focused object in the hierarchy of the modal handler, start the event
-                // bubble there
-                if (focusedObject != null && modalInput != null && focusedObject.transform.IsChildOf(modalInput.transform))
+                if (modalInput != null)
                 {
-                    if (ExecuteEvents.ExecuteHierarchy(focusedObject, eventData, eventHandler))
+                    // If there is a focused object in the hierarchy of the modal handler, start the event bubble there.
+                    if (focusedObject != null && focusedObject.transform.IsChildOf(modalInput.transform))
                     {
-                        return;
+                        if (ExecuteEvents.ExecuteHierarchy(focusedObject, eventData, eventHandler))
```

Comment:
All of these calls to ExecuteHierarchy are checking the wrong thing.

The if shouldn't be on the result of ExecuteHierarchy, it should be done on eventData.used AFTER the calls to ExecuteHierarchy. ExecuteHierarchy just returns whatever game object handled the event (if any), but that doesn't mean the event was used.

This does imply that we would need to make sure that input handlers call the Use function when necessary, otherwise we'll run into other issues sicne this changes the current behavior of InputManager